drivers/spi/spi_context.h

```diff
@@ -245,6 +245,25 @@ bool spi_context_rx_on(struct spi_context *ctx)
 	return !!(ctx->rx_buf || ctx->rx_len);
 }
 
+static ALWAYS_INLINE
+int spi_context_get_longest_single_xfer_buffers(struct spi_context *ctx,
+						void **tx_buf,
+						void **rx_buf)
+{
+	*tx_buf = ctx->tx_buf;
+	*rx_buf = ctx->rx_buf;
+
+	if (ctx->tx_len == 0) {
```

Comment:
If tx_len = 0 and rx_len 5 your code would returns 0.
In that case value 5 should be returned cause there are still bytes in tx buffer that needs transaction. I am taking min but not 0 if one of len equals 0.